text/0000-deprecate-fromstr.md

```diff
@@ -0,0 +1,81 @@
+- Feature Name: deprecate_fromstr
+- Start Date: 2020-05-11
+- RFC PR: TBD
+- Rust Issue: TBD
+
+# Summary
+[summary]: #summary
+
+Deprecate [`FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html) in favor of [`TryFrom<&str>`](https://doc.rust-lang.org/std/convert/trait.TryFrom.html)
+
+# Motivation
+[motivation]: #motivation
+
+`FromStr` was created when `TryFrom` did not exist, `FromStr` is now superfluous.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Basic implementation:
+```rust
+use std::convert::TryFrom;
+use std::num::ParseIntError;
+
+#[derive(Debug, PartialEq)]
+struct Point {
+    x: i32,
+    y: i32,
+}
+
+impl From<&str> for Point {
+    type Error = ParseIntError;
+
+    fn try_from(s: &str) -> Result<Self, Self::Error> {
+        let coords: Vec<&str> = s
+            .trim_matches(|p| p == '(' || p == ')')
+            .split(',')
+            .collect();
+
+        let x_fromstr = i32::try_from(coords[0])?;
+        let y_fromstr = i32::try_from(coords[1])?;
+
+        Ok(Point {
+            x: x_fromstr,
+            y: y_fromstr,
+        })
+    }
+}
+```
+
+Example:
+```rust
+let p = Point::try_from("(1,2)");
+assert_eq!(p.unwrap(), Point { x: 1, y: 2 })
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+- Replace all `FromStr` implementation in `std` with the corresponding `TryFrom<&str>`
+- Rewrite `str::parse()` to use `TryFrom<&str>`
```

Comment:
I don't think this is possible. This is stable code that must continue compiling:

```rust
fn my_parse<T: std::str::FromStr>(v: i32) -> T {
    match v.to_string().parse() {
        Ok(v) => v,
        Err(_) => panic!()
    }
}
```